hub/tests/test_files.py

```diff
@@ -0,0 +1,239 @@
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata2",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata2\\test2mp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata3",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testmp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testtxt.txt",
+    ]
+
+    # 실행시켰을 때 결과
+    directory = (
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata"
+    )
+    filename = None
+    recursive = True
+    extension = None
+    files = search_file(directory, filename, recursive, extension)
+
+    # 결과 비교
+    assert (
+        __files == files
+    ), f"Couldn't read all file list on {directory} (include subfolders)"
+
+
+def test_searchfile_mp4():
+    """
+    test returning the all .mp4 files list (include subfolders' one)
+    폴더 안 모든 mp4 파일 리스트 출력(하위 폴더 포함)
+    """
+    # 정상적으로 작동해야 하는 결과
+    __files = [
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\test2mp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testmp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata2\\test2mp4.mp4",
+    ]
+
+    # 실행시켰을 때 결과
+    directory = (
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata"
+    )
+    filename = None
+    recursive = True
+    extension = "mp4"
+    files = search_file(directory, filename, recursive, extension)
+
+    # 결과 비교
+    assert __files == files, f"Couldn't read all {extension} file list on {directory}"
+
+
+def test_searchfile_parentfolder():
+    """
+    test returning the all files list (NOT include subfolders' one)
+    폴더 안 모든 파일 리스트 출력(하위 폴더 미포함)
+    """
+    # 정상적으로 작동해야 하는 결과
+    __files = [
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\test2mp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata2",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata3",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testmp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testtxt.txt",
+    ]
+    # 실행시켰을 때 결과
+    directory = (
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata"
+    )
+    filename = None
+    recursive = False
+    extension = None
+    files = search_file(directory, filename, recursive, extension)
+
+    # 결과 비교
+    assert __files == files, f"Couldn't read all file list on {directory}"
```

Comment:
```suggestion
    {ERROR_NAME}_MSG = f"Couldn't read all file list on {directory}"
    assert __files == files, {ERROR_NAME}_MSG
```
이런식으로 에러가 발생하는 부분에 대해서 좀 더 명확히 나타낼 수 있으면 좋을 것 같습니다

Reply:
에러 메세지 수정했습니다~